back/scripts/enrichment/communities_enricher.py

```diff
@@ -195,3 +198,65 @@ def calculate_averages(
         )
 
         return communities
+
+    @classmethod
+    def uniformiser_noms(cls, communities: pl.DataFrame) -> pl.DataFrame:
+        import re
+
+        """
+        Uniformise les noms des collectivités selon leur type :
+        - COM -> "COMMUNE X"
+        - DEP -> "DEPARTEMENT X"
+        - REG -> "REGION X"
+        """
+
+        # Préparer le pattern pour les régions
+        region_prefixes = [
+            r"CONSEIL\s+REGIONAL\s+DE\s+LA",
+            r"COLLECTIVITE\s+TERRITORIALE\s+DE",
+            r"COLLECTIVITE\s+EUROPEENNE\s+DE",
+            r"COLLECTIVITE\s+DE",
+            r"REGION\s+DES",
+        ]
+        region_pattern = "|".join(region_prefixes)
+
+        def nettoyer_nom(nom: str, type_: str) -> str:
+            if nom is None:
+                return None
+            nom = str(nom).upper().strip()
+
+            if type_ == "COM":
+                nom = re.sub(r"^VILLE\s+DE\s+", "", nom).strip()
```

Comment:
Vu que l'on va itérer sur tous les éléments de la colonne, c'est intéressant de compiler les regex avant la boucle pour gagner en performance.

Quelque chose comme ça :
```py
regex_com = re.compile(r"^VILLE\s+DE\s+", flags=re.IGNORECASE)
def nettoyer_nom():
    if type_ == "COM":
        regex_com.sub("", nom).strip()
```
C'est valable également pour les autres.